setup.py

```diff
@@ -30,6 +30,49 @@
 import unittest
 from setuptools import setup
 import setuptools.command.build_py
+import distutils.version
+import pip
+
+
+# Make sure the environment contains an up to date enough version of pip.
+PIP_VERSION = pip.__version__
+REQUIRED_PIP_VERSION = "6.0.0"
+if (
+        distutils.version.StrictVersion(PIP_VERSION)
+        <= distutils.version.StrictVersion(REQUIRED_PIP_VERSION)
+):
+    print(
+        "Your pip version is: '{}', OpenTimelineIO requires at least "
+        "version '{}'.  Please update setuptools by running: "
+        "pip install -U pip".format(
+            PIP_VERSION,
+            REQUIRED_PIP_VERSION,
+        )
+    )
+    sys.exit(1)
+
+
+# Make sure the environment contains an up to date enough version of setuptools.
+try:
+    import setuptools.version
+    SETUPTOOLS_VERSION = setuptools.version.__version__
+except ImportError:
+    SETUPTOOLS_VERSION = setuptools.__version__
+
+REQUIRED_SETUPTOOLS_VERSION = '20.5.0'
+if (
+    distutils.version.StrictVersion(SETUPTOOLS_VERSION)
+    <= distutils.version.StrictVersion(REQUIRED_SETUPTOOLS_VERSION)
+):
+    print(
+        "Your setuptools version is: '{}', OpenTimelineIO requires at least "
+        "version '{}'.  Please update setuptools by running: "
+        "pip install -U setuptools".format(
+            SETUPTOOLS_VERSION,
+            REQUIRED_SETUPTOOLS_VERSION,
+        )
+    )
+    sys.exit(1)
```

Comment:
Could the pip and setuptools version requirements be designated by `install_requires`? What's the reason for special handling these two?

Reply:
Ah, thought it may be something like that. In that case this makes total sense!